### DISCUSSION TITLE
「レイアウト」の質問

### DISCUSSION THREAD

**Discussion by tokuda109** [vuejs-jp/learn.nuxt.com#81] [ハンズオン Q&A] [answered]
https://github.com/vuejs-jp/learn.nuxt.com/blob/main/content/2.concepts/6.layout/index.md で何か困ったことや質問があれば遠慮なく質問して下さい

**Comment by GoodDogWaffle** [comment]
レイアウトに処理を書いてはいけませんか？書いた場合、pageのonMountedとどちらが早く走りますか？

**Reply by ubugeeei** [reply, marked as answer]
もちろん書いてよいです！
(そのレイアウトを利用する) 全てのページの共通処理，みたいなものを書いておくことができます．

レイアウト，ページ，と名前はついていますが，結局はただの親子コンポーネントなので，子供の方 (page) の onMounted が先に走ります！

**Reply by atlansien** [reply]
基本的には複雑な処理などを書くのは推奨されていないと思っています
https://nuxt.com/docs/guide/directory-structure/layouts
公式にも
> 一般的な UI パターンを再利用可能なレイアウトに抽出するためのレイアウト フレームワークを提供します。

という記述になります
